packages/components/src/components/hds/app-header/index.hbs

```diff
@@ -0,0 +1,28 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div class={{this.classNames}} ...attributes>
+  {{yield to="logo"}}
+
+  {{#if (not this.isDesktop)}}
+    <Hds::AppHeader::MenuButton
+      @onClickToggle={{this.onClickToggle}}
+      @isOpen={{this.isOpen}}
+      @menuContentId={{this.menuContentId}}
+    />
+  {{/if}}
+
+  {{#if this.showItems}}
```

Comment:
I was thinking: is there maybe a way to "hide" the items when in "mobile" view (eg. using max-height + overflow hidden?) instead of removing them from the DOM? This _may_ also "solve" the problem of the `aria-controls` for the `MenuButton` below, which at this point could simply be hidden to screen readers (which have always access to the entire menu actions, it's only that they change visual presentation for the two modes, desktop and mobile)

Reply:
I think we could, I was following the pattern of our `Dropdown` component which actually removes the content when it's not displayed.

@MelSumner what would be the best practice in terms of accessibility?

Reply:
I don't recommend having duplicate content in the browser at the same time, even if some is hidden. 

- hiding interactive elements is something we've indicated folks should avoid (we should follow our own guidance)
- some browsers will expose interactive elements in the DOM and ignore anything that indicates that it shouldn't be exposed (_because_ it is an interactive element); since there are no agreed upon rules or specification for exemptions like this, they are implemented by browsers at will (and inconsistently)

Recommendation:

- Wherever possible, keep the markup the same and style for different viewports with media queries in CSS (I thought we decided on a "mobile-first" CSS approach but that is also my recommendation, I've found it easier to maintain in the long run)
- Render the correct element content to the DOM based on context

Reply:
I have since updated this to keep the container of the dropdown items in the DOM but remove the content from the DOM when it is hidden/shown as I thought you had recommended.

There isn't duplicate content currently. CSS is used to style the AppHeader content as a dropdown menu in "mobile" view vs. laying the content out within the AppHeader bar in "desktop" view.

Reply:
I'm going to look at hiding menu content using CSS vs. rendering/not rendering in phase 1 1/2